olmo/optim.py

```diff
@@ -788,6 +789,22 @@ def get_lr(self, initial_lr: float, step: int, max_steps: int) -> float:
         return initial_lr
 
 
+@dataclass
+class CosLinearEnvelope(Scheduler):
+    "Pointwise product of cosine schedule and linear decay; useful during annealing."
+    t_max: Optional[int] = None
+
+    def get_lr(self, initial_lr: float, step: int, max_steps: int) -> float:
+        eta_min = 0.0
```

Comment:
Consider setting this to `initial_lr * alpha_f`, like with the cosine and linear schedulers. Warm up would probably be nice too.

Reply:
If you use `alpha_f`, you may need to update the logic for computing the envelope.

Reply:
Changed, lmk if this looks good.